data-science-onramp/data-cleaning/clean_test.py

```diff
@@ -38,34 +40,72 @@
         },
     },
 }
+DATAPROC_JOB = {    # Dataproc job configuration
+    "placement": {"cluster_name": DATAPROC_CLUSTER},
+    "pyspark_job": {
+        "main_python_file_uri": f"gs://{BUCKET_NAME}/{BUCKET_BLOB}",
+        "args": [BQ_TABLE, BUCKET_NAME, "--dry-run"],
+        "jar_file_uris": ["gs://spark-lib/bigquery/spark-bigquery-latest_2.12.jar"],
+    },
+}
+
+
+@pytest.fixture(autouse=True)
+def setup_and_teardown_table():
+    bq_client = bigquery.Client()
+
+    # Create dataset and load table
+    dataset = bigquery.Dataset(BQ_DATASET)
+    dataset = bq_client.create_dataset(dataset)
+
+    # Load table from dataframe
+    df = pd.read_csv(CSV_FILE)
+    job_config = bigquery.LoadJobConfig(
+        autodetect=True,
+        write_disposition="WRITE_TRUNCATE"
+    )
+    operation = bq_client.load_table_from_dataframe(df, BQ_TABLE, job_config=job_config)
+
+    # Wait for job to complete
+    operation.result()
+
+    yield
+
+    # Delete dataset
+    bq_client.delete_dataset(BQ_DATASET, delete_contents=True)
 
 
 @pytest.fixture(autouse=True)
 def setup_and_teardown_cluster():
-    # Create cluster using cluster client
+    # Create Dataproc cluster using cluster client
     cluster_client = dataproc.ClusterControllerClient(
         client_options={"api_endpoint": f"{CLUSTER_REGION}-dataproc.googleapis.com:443"}
     )
-    operation = cluster_client.create_cluster(PROJECT, CLUSTER_REGION, CLUSTER_DATA)
+    operation = cluster_client.create_cluster(PROJECT_ID, CLUSTER_REGION, CLUSTER_CONFIG)
 
     # Wait for cluster to provision
     operation.result()
 
     yield
 
     # Delete cluster
-    operation = cluster_client.delete_cluster(PROJECT, CLUSTER_REGION, DATAPROC_CLUSTER)
+    operation = cluster_client.delete_cluster(
+        PROJECT_ID,
+        CLUSTER_REGION,
+        DATAPROC_CLUSTER,
+        timeout=300
```

Comment:
I'm a bit concerned about this timeout 300 parameter. Why was it added? What happens after 300 seconds (milliseconds? minutes?) if the delete operation is not complete?

Reply:
This sets the timeout to 300 seconds. The default timeout didn't seem long enough because the tear down function would fail about 50% of the time. Increasing the timeout fixed this issue.

Reply:
If the timeout is set, does the operation consistently take 300 seconds, or does it stop once the operation is complete?

Reply:
It stops once the operation is complete.